components/tidb_query/src/codec/mysql/duration.rs

```diff
@@ -666,6 +667,37 @@ impl Duration {
         }
         buf
     }
+
+    pub fn from_i64(ctx: &mut EvalContext, mut n: i64, fsp: u8) -> Result<Duration> {
```

Comment:
I think it is better to remove `ctx`  and return err if there is.
we  handle it in `convert.rs`,
this will make  the handle of  overflow and truncated err clear